vertx-grpc-server/src/main/java/io/vertx/grpc/server/impl/GrpcServerRequestImpl.java

```diff
@@ -91,4 +104,44 @@ public GrpcServerResponse<Req, Resp> response() {
   public HttpConnection connection() {
     return httpRequest.connection();
   }
+
+  @Override
+  public void handle(Buffer chunk) {
+    if (notGrpcWebText()) {
+      super.handle(chunk);
+      return;
+    }
+    if (grpcWebTextBuffer == EMPTY_BUFFER) {
+      if ((chunk.length() & 0b11) == 0) {
+        // Content length is divisible by four, so we decode it immediately
+        super.handle(Buffer.buffer(DECODER.decode(chunk.getBytes())));
+      } else {
+        grpcWebTextBuffer = chunk.copy();
+      }
+      return;
+    }
+    bufferAndDecode(chunk);
+  }
+
+  private boolean notGrpcWebText() {
+    return grpcWebTextBuffer == null;
+  }
+
+  private void bufferAndDecode(Buffer chunk) {
+    grpcWebTextBuffer.appendBuffer(chunk);
+    int len = grpcWebTextBuffer.length();
+    // Decode base64 content as soon as we have more bytes than a multiple of four.
+    // We could instead wait for the buffer length to be a multiple of four,
+    // But then in the worst case we may have to buffer the whole request.
+    int maxDecodable = len & ~0b11;
+    if (maxDecodable == len) {
+      Buffer decoded = Buffer.buffer(DECODER.decode(grpcWebTextBuffer.getBytes()));
```

Comment:
we should use io.netty.handler.codec.base64.Base64 here that decodes a ByteBuf to a ByteBuf which avoids ByteBuf -> byte[] -> byte[] -> ByteBuf